### PR TITLE
feat(node): `Duplex.fromWeb`

### PR DESCRIPTION
**Pull request by crowlKats**
Needed for #2061
not sure where to put `isReadableStream` & `isWritableStream`. 
Also it seems node has no test for this

### CODE REVIEW

File: node/internal/streams/duplex.mjs [hunk 1]
```diff
@@ -2,21 +2,27 @@
 // Copyright Joyent and Node contributors. All rights reserved. MIT license.
 // deno-lint-ignore-file
 
+import { validateObject, validateBoolean } from "../validators.mjs";
 import { createDeferredPromise } from "../util.mjs";
 import { destroyer } from "./destroy.mjs";
 import { isBlob } from "../blob.mjs";
+import { Buffer } from "../../buffer.mjs";
```
Comment (kt3k) [inline node/internal/streams/duplex.mjs:9]: typo?
```suggestion
import { Buffer } from "../../buffer.ts";
```

**Comment by kt3k** [comment]
It looks like they are testing the internal utils (such as `newStreamDuplexFromReadableWritablePair`) instead of testing public interfaces directly https://github.com/nodejs/node/pull/39134/files#diff-d8ce88f0cd20640d6cdffb550abcd3a8fb9b4e543cc04bc87c2d827802e7dc60

Is it possible to somehow borrow these test cases to fix into our situation?

**Comment by crowlKats** [comment]
@kt3k no idea. looking through those tests, they use an internal symbol for web streams called `kState`. we dont have that, so dont think we can even run those tests...

**Comment by kt3k** [comment]
Then maybe let's invent something very simple for now. It also will be tested in the implementation of #2061

**Review by bartlomieju — APPROVED** [review]
I agree, let's land it now.

**Pull request merged**